compiler/src/Language/Bond/Codegen/Java/Class_java.hs

```diff
@@ -37,23 +228,157 @@ package #{javaPackage};
 
-    @Override
-    public com.microsoft.bond.StructBondType<? extends BondSerializable> getBondType() {
-        return BOND_TYPE;
+        static final class __StructBondTypeBuilderImpl extends com.microsoft.bond.StructBondType.StructBondTypeBuilder<#{declName}> {
+            #{makeGenericTypeMethodImpl}
+
+            @Override
+            public final int getGenericTypeParameterCount() {
+                return #{length declParams};
+            }
+
+            @Override
+            protected final StructBondType<#{declName}> buildNewInstance(BondType<?>[] genericTypeArguments) {
+                #{buildNewInstanceMethodBody}
+            }
+
+            static void register() {
+                registerStructType(#{declName}.class, new __StructBondTypeBuilderImpl());
+            }
+        }
+
+        #{doubleLineSep 2 fieldDescriptorFieldDecl structFields}
+
+        #{structBondTypeImplConstructorDecl} {
+            #{structBondTypeImplConstructorBody}
+        }
+
+        @Override
+        protected final void initialize() {
+            #{newlineSep 3 genericTypeArgumentVariableDecl declParamsWithIndex}
+            #{newlineSep 3 fieldDescriptorFieldInit structFields}
+            super.initializeBaseAndFields(#{initializeBaseAndFieldsBaseParam}#{initializeBaseAndFieldsFieldsParam});
+        }
+
+        @Override
+        public final java.lang.Class<#{declName}#{params}> getValueClass() {
+            return (java.lang.Class<#{declName}#{params}>) (java.lang.Class<?>) #{declName}.class;
+        }
+
+        @Override
+        public final #{declName}#{params} newInstance() {
+            #{newInstanceMethodBody}
+        }
+
+        @Override
+        protected final void serializeStructFields(com.microsoft.bond.BondType.SerializationContext context, #{declName}#{params} value) throws java.io.IOException {
+            #{newlineSep 3 serializeStructField structFields}
+        }
+
+        @Override
+        protected final void deserializeStructFields(com.microsoft.bond.BondType.TaggedDeserializationContext context, #{declName}#{params} value) throws java.io.IOException {
+            #{newlineSep 3 deserializeStructFieldDeclareBooleans structFields}
+            while (readField(context)) {
+                switch (context.readFieldResult.id) {
+                    #{newlineSep 5 deserializeStructFieldProcessFieldCases structFields}
+                }
+            }
+            #{newlineSep 3 deserializeStructFieldVerifyFields structFields}
+        }
+
+        @Override
+        public final void initializeStructFields(#{declName}#{params} value) {
+            #{newlineSep 3 initializeStructField structFields}
+        }
+    }
+
+    #{bondTypeStaticVariableDecl}
+
+    public static void initializeBondType() {
+        __StructBondTypeImpl.__StructBondTypeBuilderImpl.register();
+    }
+
+    static {
+        initializeBondType();
```

Comment:
No, this function needs to be public. Static block alone aren't sufficient (consider complex/circular class dependencies in multi-threaded environments). This function is called by the code (alt. we could move this to the constructor with a small per-instance overhead). 

In fact, we could remove the static block.

---
In reply to: [124944189](https://github.com/Microsoft/bond/pull/517#discussion_r124944189) [](ancestors = 124944189)